src/AnalysisHelper.cc

```diff
@@ -344,21 +344,39 @@ void AnalysisHelper::prepareSamplesHistos()
                 for (uint ivar = 0; ivar < variables_.size(); ++ivar)
                 {
                     string varName = variables_.at(ivar);
+                    string selName = selections_.at(isel).getName();
+                    string sel_var = selName + ":" + varName;
 
                     varcoll.append(varName, Sample::systColl());
                     Sample::systColl& systcoll = varcoll.back();
                     
                     bool   hasUserBinning = cutCfg_->hasOpt(Form("binning::%s", varName.c_str()));
-		    int    nbins = -1;
+                    bool   hasUserBinning_thisSel = cutCfg_->hasOpt(Form("binning::%s", sel_var.c_str()));
+                    int    nbins = -1;
                     float  xlow = -1.;
                     float  xup = -1.;
                     float* binning = 0;
 
-                    if (hasUserBinning)
+                    if (hasUserBinning | hasUserBinning_thisSel )
```

Comment:
This should be || not |

Reply:
right my bad

Reply:
Now it should be fixed. 
I also forgot to add the `hasUserBinning_thisSel` when deleting the `binning` object.
Both should be fixed in 2b430b0